### PR TITLE
add ability to register runner with github token

### PR DESCRIPTION
**Pull request by PH1P5**
Hi @SanderKnape ,

this PR adds the ability to register a runner with github token. This is quite handy since a runner deployment is not tied to a personal access token from a certain contributor.

I think that solves #19 

Please have a look at the changes and merge it, or give feedback if there are any changes needed.

Thank you for your effort

p.s.: added a more understandable error message if non-root tries to install additional packages

fyi: @hennr , @5nafu , @fsperling

### CODE REVIEW

**Comment by SanderKnape** [comment]
Thanks @PH1P5, I'll take a good look later today. Just out of curiosity: where would you get the token you pass to the runner? The token currently generated in the entrypoint is only valid for an hour which is why I choose to pass the PAT to the runner.

**Comment by PH1P5** [comment]
Hi @SanderKnape, not sure if it is called "repository token" but it will be generated in the Actions / Runner section in the repo settings in the description for how to add a runner:

Settings -> Actions -> Runners -> Add runner

**Comment by SanderKnape** [comment]
Right, that token is only valid for one hour (documented here for example: https://docs.github.com/en/rest/reference/actions#create-a-registration-token-for-a-repository). So you have another process that fetches the token before starting the container?

I'm fine with adding this btw - just curious what the workflow would look like.

File: entrypoint.sh [hunk 1]
```diff
@@ -1,23 +1,42 @@
 #!/bin/sh
 
 if [ -n "${ADDITIONAL_PACKAGES}" ]; then
-    TO_BE_INSTALLED=$(echo ${ADDITIONAL_PACKAGES} | tr "," " " )
-    echo "Installing additional packages: ${TO_BE_INSTALLED}"
-    sudo apt-get update && sudo apt-get install -y ${TO_BE_INSTALLED} && sudo apt-get clean
+    if [ $(id -u) = 0 ]; then
```
Comment (SanderKnape) [inline entrypoint.sh]: The `github` user has sudo permissions, so there is no need to be root.

Comment (PH1P5) [reply entrypoint.sh]: True, i will adjust this within the next days. Regarding fetch token process - currently we deploy our runner manually  so we need to refresh the token in the environments secrets on redeploy.

**Comment by 5nafu** [comment]
When the GitHub runner is configured with `./config.sh`, there will be a couple of dot files created. One of them is a `.credentials` (and `.credentials-rsaparams`) file where an oauth token with (longer|unlimited) validity is stored.

Maybe it would be possible to allow these dot files to be part of a mount/persistent volume to be copied from/to. And the entrypoint to check if these files exist. If they exist, configuration could be skipped and the appropriate files copied back to the root directory.

File: entrypoint.sh (hunk 1 above)
Comment (PH1P5) [reply entrypoint.sh]: For this PR i rolled back changes regarding root guard since it does not match initial intent.

**Comment by PH1P5** [comment]
@5nafu that sounds promising. Shall we work on this together to create another PR?

**Comment by SanderKnape** [comment]
Thanks for the contribution @PH1P5 !

**Pull request merged**